azurelinuxagent/common/cgroups.py

```diff
@@ -556,16 +564,21 @@ def setup(suppress_process_add=False):
             try:
                 CGroups._osutil.mount_cgroups()
                 if not suppress_process_add:
+                    # Creates /sys/fs/cgroup/{cpu,memory}/WALinuxAgent wrapper cgroup
                     CGroups._setup_wrapper_groups()
                     pid = int(os.getpid())
                     if not CGroups.is_systemd_manager():
-                        cg = CGroups.for_extension(AGENT_NAME)
-                        logger.info("Add daemon process pid {0} to {1} cgroup".format(pid, cg.name))
+                        # Creates /sys/fs/cgroup/{cpu,memory}/WALinuxAgent/WALinuxAgent cgroup
+                        cg = CGroups.for_extension(AGENT_CGROUP_NAME)
+                        logger.info("Daemon process id {0} is tracked in cgroup {1}".format(pid, cg.name))
                         cg.add(pid)
                         cg.set_limits()
                     else:
-                        cg = CGroups.for_systemd_service(AGENT_NAME)
-                        logger.info("Add daemon process pid {0} to {1} systemd cgroup".format(pid, cg.name))
+                        # When daemon is running as a service, it's called walinuxagent.service
+                        # and is created and tracked by systemd, so we don't explicitly add the PID ourselves,
+                        # just track it for our reporting purposes
+                        cg = CGroups.for_systemd_service(AGENT_CGROUP_NAME.lower() + ".service")
+                        logger.info("Daemon process id {0} is tracked in systemd cgroup {1}".format(pid, cg.name))
                         # systemd sets limits; any limits we write would be overwritten
                 status = "ok"
```

Comment:
Can you please add a better status message? For a successful setup.

Reply:
I agree that it's not very descriptive, I can change it but since this status is sent to our telemetry logs, we have to be mindful about querying for both the old and new message now.